cmip6_downscaling/methods/bcsd.py

```diff
@@ -0,0 +1,249 @@
+    """
+
+    """
+
+    TODO: do we want to assign this to a class? write out? probably
+    """
+    # TODO: add in functionality to label the obs/spatial anomalies by grid name as opposed to GCM
+    #  since it will just help us not repeat the coarsening/spatial anomalies step
+    if rerun:
+        obs_ds = load_obs(obs_id, variable, time_period=slice(train_period_start, train_period_end))
+
+        gcm_one_slice = load_cmip(return_type='xr', variable_ids=[variable]).isel(time=0)
+        # calculate and write out the coarsened version of obs dataset to match the gcm
+        # (will be used in training)
+        coarse_obs = regrid_dataset(
+            obs_ds, gcm_one_slice, variable='tasmax', connection_string=connection_string
+        )
+        write_dataset(coarse_obs, coarse_obs_path)
+        # calculate the seasonal cycle (ntime = 12) of spatial anomalies due to interpolating
+        # the coarsened obs back to its original resolution and write it out (will be used in postprocess)
+        spatial_anomolies = get_spatial_anomolies(coarse_obs, obs_ds, variable, connection_string)
+        write_dataset(spatial_anomolies, spatial_anomolies_path, chunks_dims=('month',))
+
+    return coarse_obs_path, spatial_anomolies_path
+
+
+def prep_bcsd_inputs(
+    coarse_obs_path: str,
+    gcm: str,
+    scenario: str,
+    obs_id: str,
+    train_period_start: str,
+    train_period_end: str,
+    predict_period_start: str,
+    predict_period_end: str,
+    variable: str,
+) -> tuple[str, str, str]:
+    """Prepare the inputs to be fed into the training and fitting of the model. Largely
+    this converts any datasets chunked in time into ones chunked in space since
+    the models are pointwise and need the full timeseries to be performant.
+
+    Parameters
+    ----------
+    coarse_obs_path : str
+        Path to read coarsened obs zarr dataset (e.g. 'az://bucket/out.zarr')
+    gcm : str
+        Name of GCM
+    obs_id : str
+        Name of obs dataset (currently only supports 'ERA5')
+    train_period_start : str
+        Date for training period start (e.g. '1985')
+    train_period_end : str
+        Date for training period end (e.g. '2015')
+    predict_period_start : str
+        Date for prediction period start (e.g. '2070')
+    predict_period_end : str
+        Date for prediction period end (e.g. '2099')
+    variable : str
+        Variable of interest in CMIP conventions (e.g. 'tasmax')
+
+    Returns
+    -------
+    y_rechunked_path, X_train_rechunked_path, X_predict_rechunked_path : str, str, str
+        Paths to to write y, X_train, and X_predict ready for eventual use by model
+    """
+    # load in coarse obs as xarray ds
+    [y] = load_paths([coarse_obs_path])
+    # TODO: add xarray_schema test here for chunking - if the chunking fails then try rechunking
+    # if passes skip
+    y_rechunked, y_rechunked_path = rechunk_zarr_array(
```

Comment:
nit: It seems like y_rechunked and x_train_rechunked (line 142) are both unused.

Reply:
good catch! also x_predict_rechunked. i've changed them to `_`